lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -476,19 +474,21 @@ def getvm(self, name=None, uuid=None, folder=None):
         if uuid:
             vm = self.content.searchIndex.FindByUuid(uuid=uuid, vmSearch=True)
         elif folder:
-            # Build the absolute folder path to pass into the search method
-            if not self.params['folder'].startswith('/'):
-                self.module.fail_json(msg="Folder %(folder)s needs to be an absolute path, starting with '/'." % self.params)
-            searchpath = '%(datacenter)s%(folder)s' % self.params
+            # searchpaths do not need to be absolute
+            searchpath = self.params['folder']
```

Comment:
Can we move `getvm` method to `vmware.py` so that `vmware_guest_facts` or other modules could use it?